cpp/tests/prims/mg_per_v_transform_reduce_incoming_outgoing_e.cu

```diff
@@ -71,18 +71,53 @@ struct e_op_t {
   }
 };
 
+template <typename T>
```

Comment:
Would it make sense to pull this code (`compare_scalar` and `struct comparator`)  out into a common test utility?  It feels like this would be useful in other tests as well.

Reply:
Yes, I will make this in a MG C++ test cleanup PR. This PR is already large enough :-)